x/programs/rust/examples/single_nft/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+//! A basic ERC-721 compatible contract.
+//! The program serves as a non-fungible token with the ability to mint and burn.
+//! Only supports whole units with no decimal places.
+//!
+//! The NFT must support the common NFT metadata format.
+//! This includes the name, symbol, and URI of the NFT.
+use metadata::Nft;
+use wasmlanche_sdk::{program::Program, public, state_keys, types::Address};
+
+pub mod example;
+pub mod metadata;
+
+const NAME: &str = "My NFT";
+const SYMBOL: &str = "MNFT";
+const TOTAL_SUPPLY: u64 = 1;
+
+/// The program storage keys.
+#[state_keys]
+enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
+    /// The name of the token. Key prefix 0x1.
+    Name,
+    /// The symbol of the token. Key prefix 0x2.
+    Symbol,
+    /// Metadata of the token. Key prefix 0x3.
+    Metadata,
+    /// Balance of the NFT token by address. Key prefix 0x4(address).
+    Balance(Address),
+    /// Counter -- used to keep track of total NFTs minted. Key prefix 0x5.
+    Counter,
+    /// Owner -- used to keep track of the owner of each NFT. Key prefix 0x6.
+    Owner,
+}
+
+/// Initializes the NFT with all required metadata.
+/// This includes the name, symbol, image URI, owner, and total supply.
+/// Returns true if the initialization was successful.
+#[public]
+pub fn init(program: Program) -> bool {
```

Comment:
`Name` `Symbol` and `TotalSupply` should be params on init. I think the idea is these examples can be pulled of the shelf and deployed/tested. In this case user must change the code.